go/src/koding/klient/machine/dynamic.go

```diff
@@ -192,10 +192,14 @@ func (dc *DynamicClient) cron() {
 
 // tryUpdate uses client builder to ping the machine and updates dynamic client
 // if machine address changes.
-func (dc *DynamicClient) tryUpdate(addr *Addr) {
+func (dc *DynamicClient) tryUpdate(addr *Addr, islog *bool) {
 	stat, a, err := dc.opts.Builder.Ping(dc.opts.AddrFunc)
 	if err != nil {
-		dc.log.Warning("Machine ping error: %s", err)
+		if *islog {
```

Comment:
I would rather add a ThrottledLogger type and used it globally - the logic whether to display a log belongs to Logger not a DynamicClient.

```go
type ThrottledLogger struct {
   Throttle time.Duration
   Log      logging.Logger
}

func (th *ThrottledLogger) Error(fmt string, args ...interface{}) {
    th.logOnce(th.log.Error, fmt, args...)
}

...
```

This way we could get rid of the bool patching here.